pulsar/consumer_partition.go

```diff
@@ -1673,11 +1680,8 @@ func (pc *partitionConsumer) reconnectToBroker() {
 			return
 		}
 
-		if pc.options.backoffPolicy == nil {
-			delayReconnectTime = defaultBackoff.Next()
-		} else {
-			delayReconnectTime = pc.options.backoffPolicy.Next()
-		}
+		delayReconnectTime = bo.Next()
+		totalDelayReconnectTime += delayReconnectTime
```

Comment:
We also need to reset the `delayReconnectTime` once connected after several retries in 1692.

Reply:
After a successful reconnection at line 1699, the function will exit, and the `delayReconnectTime` variable will be initialized during the next reconnection. 

Therefore, there is no need to reset `delayReconnectTime` here.